github-stat-tools/github-pending.py

```diff
@@ -0,0 +1,124 @@
+# the metric, like Prometheus or Loki do.  Instead, we feed all of them as
+# separate values in one input, with the hope that they all get the same
+# time stamp
+def backend_zabbix(host, basekey, values):
+    zabbix_command = ['zabbix_sender', '-z', 'm.openssl.org', '-i', '-']
+    zabbix_lines = [f'{host} {basekey}.{k} {v}' for k,v in values.items()]
+    zabbix_input = "\n".join(zabbix_lines) + "\n"
+    if debug or dryrun:
+        prefix = 'DEBUG[backend_zabbix]: ' if debug else ''
+        intro = 'would run this command:' if dryrun else 'running this command:'
+        for l in [ intro,
+                   '',
+                   ' '.join([ *zabbix_command, '<<_____']),
+                   *zabbix_lines,
+                   '_____',
+                   '' ]:
+            print(f'{prefix}{l}', file=sys.stderr)
+
+    if not dryrun:
+        subprocess.run(zabbix_command, input=zabbix_input, text=True);
+
+# Echoing is done in a way that's similar to Prometheus / Loki input.
+# The "metric" value is treated specially, so it becomes the actual sole
+# value, while the rest of the values are indexing label values.
+def backend_echo(host, basekey, values):
+    t = now.isoformat()
+    s = f'{basekey}' + '{' + ', '.join(
+        [ f'host="{host}"',
+          *( f'{k}="{v}"' for k,v in values.items() if k != 'metric' ) ]
+    ) + '}'
+    print(f'{t}: {s} {values["metric"]}')
+
+### Main
+
+# defaults
+host = 'github.com'
+backend = 'echo'
+# blank token is fine, but you may hit API rate limiting
+git_token = ''
+
+# parse options
+parser = ArgumentParser()
+
+parser.add_argument('--backend', '-b',
+                    help='the output backend.  Accepted choices: echo, zabbix',
+                    dest='backend')
+parser.add_argument('--host',
+                    help='the github host to check',
+                    dest='host')
+parser.add_argument('--token', '-t',
+                    help='file containing github authentication token for example "18asdjada..."',
+                    dest='token')
+parser.add_argument('--debug', '-d', action='store_true', help='be noisy',
+                    dest='debug')
+parser.add_argument('--dry-run', '-n', action='store_true', help='be noisy',
+                    dest='dryrun')
+
+args = parser.parse_args()
+
+if args.backend:
+    backend = args.backend
+if args.host:
+    host = args.host
+if args.token:
+    fp = open(args.token, 'r')
+    git_token = fp.readline().strip('\n')
+debug = args.debug
+dryrun = args.dryrun
+
+# info databases
+backends = {
+    'zabbix': backend_zabbix,
+    'echo': backend_echo,
+}
+
+headers = {
+    'Accept': 'application/vnd.github+json',
+    'Authorization': git_token,
+}
+
+# Do stuff
+open_issues = search(
+    host, [ 'repo:openssl/openssl', 'type:issue', 'state:open' ], headers
+)
+open_pulls = search(
+    host, [ 'repo:openssl/openssl', 'type:pr', 'state:open' ], headers
+)
+
+backends[backend](host, 'openssl.issues.gap',
```

Comment:
I know that Python can take care of missing members of data structure and make a traceback, still I like to check what goes into (should be: "zabbix" or "echo").

Reply:
Ah, good point.  Will do

Reply:
Have a look at my latest fixup, which uses the `choices` option for `parser.add_argument`